src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2520,23 +2520,12 @@ class StubGenerator: public StubCodeGenerator {
     assert((base_offset2 % (UseCompactObjectHeaders ? 4 :
                             (UseCompressedClassPointers ? 8 : 4))) == 0, "Must be");
 
-    // cnt2 == amount of characters left to compare
-    // Check already loaded first 4 symbols
-    __ inflate_lo32(tmp3, isLU ? tmp1 : tmp2);
-    __ mv(isLU ? tmp1 : tmp2, tmp3);
-    __ addi(str1, str1, isLU ? wordSize / 2 : wordSize);
-    __ addi(str2, str2, isLU ? wordSize : wordSize / 2);
-    __ subi(cnt2, cnt2, wordSize / 2); // Already loaded 4 symbols
-
-    __ xorr(tmp3, tmp1, tmp2);
-    __ bnez(tmp3, CALCULATE_DIFFERENCE);
-
     Register strU = isLU ? str2 : str1,
              strL = isLU ? str1 : str2,
              tmpU = isLU ? tmp2 : tmp1, // where to keep U for comparison
              tmpL = isLU ? tmp1 : tmp2; // where to keep L for comparison
 
-    if (AvoidUnalignedAccesses && (base_offset1 % 8) == 0) {
+    if (AvoidUnalignedAccesses && (base_offset1 % 8) != 0) {
       // Load another 4 bytes from strL to make sure main loop is 8-byte aligned
```

Comment:
Nit: You might want to update this code comment removing the word `another`.
`// Load another 4 bytes from strL to make sure main loop is 8-byte aligned`
=>
`// Load 4 bytes from strL to make sure main loop is 8-byte aligned`

Reply:
Yes, fixed.